providers/base/bin/snap_update_test.py

```diff
@@ -0,0 +1,336 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# All rights reserved.
+#
+# Written by:
+#    Patrick Liu <patrick.liu@canonical.com>
+#    Pierre Equoy <pierre.equoy@canonical.com>
+#
+# Checkbox is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# Checkbox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox.  If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+from glob import glob
+import json
+import logging
+import os.path
+import sys
+import time
+
+from checkbox_support.snap_utils.snapd import Snapd
+
+
+def guess_snaps() -> dict:
+    """
+    Guess the names of the kernel, snapd and gadget snaps from installed snaps
+    on the system.
+
+    :return: a dict with the snap names for each snap found
+    :rtype: dict
+    """
+    snapd = Snapd()
+    installed_snaps = snapd.list()
+    snaps = {}
+    for snap in installed_snaps:
+        if snap["type"] == "kernel":
+            snaps["kernel"] = snap["name"]
+        elif snap["type"] == "gadget":
+            snaps["gadget"] = snap["name"]
+        elif snap["type"] == "snapd":
+            snaps["snapd"] = snap["name"]
+    return snaps
+
+
+def get_snap_base_rev() -> dict:
+    """
+    Retrieve the name and the base revision of each snap originally installed
+    on the system.
+
+    :return: a dict containing the snap names and their base revisions
+    :rtype: dict
+    """
+    base_snaps = glob("/var/lib/snapd/seed/snaps/*.snap")
```

Comment:
I think it's because you're *using Arch™* :laughing: 

The `/var/lib/snapd/seed/snaps/` directory contains the base revision of the snaps installed by a given ISO image.

If you installed snapd afterwards, I think it's normal not to have the `/var/lib/snapd/seed/snaps/` dir.

FYI, this is what it looks like on my desktop (Ubuntu 22.04):

```
ls /var/lib/snapd/seed/snaps/
bare_5.snap  core20_1587.snap  firefox_1635.snap  gnome-3-38-2004_112.snap  gtk-common-themes_1535.snap  snapd_16292.snap  snapd-desktop-integration_14.snap  snap-store_582.snap
```